### PR TITLE
Blu vimclutch

### PR DESCRIPTION
**Pull request by blu006**
Added a vimclutch implementation using qmk and tested all included keymaps.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [X] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/blu/vimclutch/keymaps/default/keymap.c [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+ */
+#include QMK_KEYBOARD_H
+
+enum layer_names {
+    _VC,
+    _VIM
+};
+
+enum my_keycodes {
+    USER_CLUTCH = SAFE_RANGE
```
Comment (waffle87) [inline keyboards/blu/vimclutch/keymaps/default/keymap.c:24]:
```suggestion
    USER_CLUTCH = QK_USER
```

File: keyboards/blu/vimclutch/keymaps/vc_i/keymap.c [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+ */
+#include QMK_KEYBOARD_H
+
+enum layer_names {
+    _VC,
+    _VIM
+};
+
+enum my_keycodes {
+    USER_CLUTCH = SAFE_RANGE
```
Comment (waffle87) [inline keyboards/blu/vimclutch/keymaps/vc_i/keymap.c:24]:
```suggestion
    USER_CLUTCH = QK_USER
```

File: keyboards/blu/vimclutch/readme.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+Here are some included keymaps and their modes to get you started:
+* default: voice chat and vim clutch
+* vc_i: the same as default but using `i` instead of `a`
+* vim_ai: vim clutch only, selectable between `a` and `i`
+
+Also includes a mode for single-keypress in case of "push to talk clutch"
+
+Make example for this keyboard after setting up your development environment:
+
+      make blu/vimclutch:default
```
Comment (waffle87) [inline keyboards/blu/vimclutch/readme.md:26]:
```suggestion
    make blu/vimclutch:default
      
Flashing example for this keyboard:

    make blu/vimclutch:default

## Bootloader
```

File: keyboards/blu/vimclutch/vimclutch.c [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+
+GNU General Public License for more details.
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#include "vimclutch.h"
```
Comment (waffle87) [inline keyboards/blu/vimclutch/vimclutch.c:17]:
```suggestion
#include "quantm.h"
```

File: keyboards/blu/vimclutch/vimclutch.h [hunk 5]
```diff
@@ -0,0 +1,20 @@
+/*
```
Comment (waffle87) [inline keyboards/blu/vimclutch/vimclutch.h:1]: This file should be removed.

**Review by waffle87 — CHANGES_REQUESTED** [review]

File: keyboards/blu/vimclutch/keymaps/default/keymap.c (hunk 1 above)
Comment (blu006) [reply keyboards/blu/vimclutch/keymaps/default/keymap.c:24]: oh, the constant changed.  Thank you.

File: keyboards/blu/vimclutch/readme.md (hunk 3 above)
Comment (blu006) [reply keyboards/blu/vimclutch/readme.md:26]: thanks; do you use a tool to catch extra whitespace?

File: keyboards/blu/vimclutch/vimclutch.h (hunk 5 above)
Comment (blu006) [reply keyboards/blu/vimclutch/vimclutch.h:1]: I agree.

**Comment by blu006** [comment]
Sorry, I briefly had this in draft because `qmk build` did not like my changes when I forgot to `qmk clean` before building.

File: keyboards/blu/vimclutch/vimclutch.c [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+
+GNU General Public License for more details.
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#include "quantm.h"
```
Comment (waffle87) [inline keyboards/blu/vimclutch/vimclutch.c:17]: Oops, typo on my part.
```suggestion
#include "quantum.h"
```

**Review by waffle87 — APPROVED** [review]

File: keyboards/blu/vimclutch/vimclutch.c (hunk 6 above)
Comment (blu006) [reply keyboards/blu/vimclutch/vimclutch.c]: I noticed this during my compilation as well, but forgot to commit it before asking for the re-review.

File: keyboards/blu/vimclutch/keymaps/default/keymap.c [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+        break;
+    case _VIM:
+        writePinHigh(F4);
+	writePinLow(F5);
+        break;
+    default:
+        writePinHigh(F4);
+        writePinHigh(F5);
+        break;
+    }
```
Comment (drashna) [inline keyboards/blu/vimclutch/keymaps/default/keymap.c:53]: This is the same, but much less code:
```suggestion
    writePin(F4, !layer_state_cmp(state, _VC));
    writePin(F5, !layer_state_cmp(state, _VIM));
```

File: keyboards/blu/vimclutch/keymaps/vc_i/keymap.c [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+    ),
+
+    [_VIM] = LAYOUT(
+        TO(_VC),    TO(_VIM),  USER_CLUTCH
+    )
+};
+
+/* Layer-specific lighting */
+layer_state_t layer_state_set_user(layer_state_t state) {
+    switch (get_highest_layer(state)) {
```
Comment (drashna) [inline keyboards/blu/vimclutch/keymaps/vc_i/keymap.c:40]: same as other.

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/blu/vimclutch/keymaps/vc_i/keymap.c (hunk 8 above)
Comment (blu006) [reply keyboards/blu/vimclutch/keymaps/vc_i/keymap.c:40]: Hi drashna, nice to see you as always.

I'll make the same updates to the layouts in one commit and refer to that commit.

Comment (blu006) [reply keyboards/blu/vimclutch/keymaps/vc_i/keymap.c]: Implemented and tested in hardware on https://github.com/qmk/qmk_firmware/pull/20060/commits/77f2a52d7506fd72fbfce3a829fd10ba2bf5eb3d

**Comment by blu006** [comment]
Hi guys, the changes requested were completed when I re-requested a review.

I have also tried them out on actual hardware, and they work.

File: keyboards/blu/vimclutch/info.json [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+	"manufacturer": "blu",
+	"url": "https://github.com/blu006/vimclutch_hw",
+	"maintainer": "blu006",
+	"usb": {
+		"vid": "0xFEED",
+		"pid": "0x6060",
+		"device_version": "0.0.1"
+	},
+	"processor": "atmega32u4",
+	"bootloader": "caterina",
```
Comment (drashna) [inline keyboards/blu/vimclutch/info.json:12]:
```suggestion
	"development_board": "promicro",
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

**Pull request merged**